src/viper/syntax.ml

```diff
@@ -90,8 +92,21 @@ and stmt' =
 
 and typ = (typ', info) Source.annotated_phrase
 
+
+and mut = Const | Mut
```

Comment:
Is this flag for future use? Looks like the translation ignores it.

Reply:
It's actually used in array utility functions. Check `array_acc`.
I use it for identifying which kind of an access predicate should be used. Namely `ArrayT Mut` requires write access into each location so it should be `array_acc_var` which is a macro for `forall ... acc(loc(a, i).val, write)`

Reply:
Actually I did it only for convenience. We can eliminate it and pass this information as a separate parameter into utility functions. But I put it this way.